modules/installation-cloudformation-vpc-localzone.adoc

```diff
@@ -0,0 +1,379 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws/installing-aws-localzone.adoc
+
+[id="installation-cloudformation-vpc-localzone_{context}"]
+= CloudFormation template for the VPC that uses AWS Local Zones
+
+You can use the following CloudFormation template to deploy the VPC that
+you need for your {product-title} cluster that uses AWS Local Zones.
+
+.CloudFormation template for the VPC
+[%collapsible]
+====
+[source,yaml]
+----
+AWSTemplateFormatVersion: 2010-09-09
+Description: Template for Best Practice VPC with 1-3 AZs
+
+Parameters:
+  ClusterName:
+    Type: String
+    Description: ClusterName used to prefix resource names
+  ClusterInfraId:
+    Type: String
+    Default: "unmanaged"
+    Description: ClusterInfraId used to tag required resources
+  VpcCidr:
+    AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/(1[6-9]|2[0-4]))$
+    ConstraintDescription: CIDR block parameter must be in the form x.x.x.x/16-24.
+    Default: 10.0.0.0/16
+    Description: CIDR block for VPC.
+    Type: String
+  AvailabilityZoneCount:
+    ConstraintDescription: "The number of availability zones. (Min: 1, Max: 3)"
+    MinValue: 1
+    MaxValue: 3
+    Default: 1
+    Description: "How many AZs to create VPC subnets for. (Min: 1, Max: 3)"
+    Type: Number
+  SubnetBits:
+    ConstraintDescription: CIDR block parameter must be in the form x.x.x.x/19-27.
+    MinValue: 5
+    MaxValue: 13
+    Default: 12
+    Description: "Size of each subnet to create within the availability zones. (Min: 5 = /27, Max: 13 = /19)"
+    Type: Number
+
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+    - Label:
+        default: "Network Configuration"
+      Parameters:
+      - VpcCidr
+      - SubnetBits
+    - Label:
+        default: "Availability Zones"
+      Parameters:
+      - AvailabilityZoneCount
+    ParameterLabels:
+      ClusterName:
+        default: ""
+      AvailabilityZoneCount:
+        default: "Availability Zone Count"
+      VpcCidr:
+        default: "VPC CIDR"
+      SubnetBits:
+        default: "Bits Per Subnet"
+
+Conditions:
+  DoAz3: !Equals [3, !Ref AvailabilityZoneCount]
+  DoAz2: !Or [!Equals [2, !Ref AvailabilityZoneCount], Condition: DoAz3]
+
+Resources:
+  VPC:
+    Type: "AWS::EC2::VPC"
+    Properties:
+      EnableDnsSupport: "true"
+      EnableDnsHostnames: "true"
+      CidrBlock: !Ref VpcCidr
+      Tags:
+      - Key: Name
+        Value: !Join [ "", [ !Ref ClusterName, "-vpc" ] ]
+      - Key: !Join [ "", [ "kubernetes.io/cluster/", !Ref ClusterInfraId ] ]
```

Comment:
When removing the unused parameter `ClusterInfraId`[1], we **should** set the value of this tag to unmanned.

This should something we need to ship in this version, the in-tree cloud-provider uses those tags to discover the subnets to create the default ingress controller. If the Local Zones subnets is discovered by the ingress operator on the install time, it's expected to the installation to fail. We got this issue on the begging on the research and we are working with upstream to address it on the better way, but the legacy code seems to be untouchable.

Please take a look at [1][2][3].


```suggestion
      - Key: !Join [ "", [ "kubernetes.io/cluster/unmanaged" ] ]
```

```
  ClusterInfraId:
    Type: String
    Default: "unmanaged"
    Description: ClusterInfraId used to tag required resources
```

[1] tracking the discover bug https://issues.redhat.com/browse/OCPBUGSM-46513
[2] tracking card for ALB Operator requiring to set this tag https://issues.redhat.com/browse/OCPBUGSM-46513
[3] Upstream discussion: https://kubernetes.slack.com/archives/C718BPBQ8/p1672850778127529